src/main/java/com/ays/backend/user/repository/UserRepository.java

```diff
@@ -0,0 +1,19 @@
+package com.ays.backend.user.repository;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import com.ays.backend.user.model.entities.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+/**
+ * UserRepository performing DB layer operations.
+ */
+public interface UserRepository extends JpaRepository<User, Long> {
+
+    Optional<User> findByUsername(String username);
+
+    Boolean existsByUsername(String username);
+
+    Optional<User> findByUserUUID(UUID userId);
```

Comment:
Errors out by `No property 'userUUID' found for type 'User'` due to d2759b0bedf150d9043a46d0d808c03d01d1c8af. The spec says id, not uuid, but I'm not sure if that's short for uuid or sequential id. I think we should go with uuid. In any case I think it's a good idea to add @Id onto the field.